tsl/src/continuous_aggs/invalidation.c

```diff
@@ -314,27 +314,48 @@ invalidation_hyper_log_add_entry(int32 hyper_id, int64 start, int64 end)
  * hypertable.
  */
 void
-invalidation_add_entry(const Hypertable *ht, ContinuousAggHypertableStatus caggstatus,
-					   int32 entry_id, int64 start, int64 end)
+continuous_agg_invalidate_raw_ht(const Hypertable *raw_ht, int64 start, int64 end)
 {
-	Assert(ht != NULL);
+	continuous_agg_invalidate_mat_ht(raw_ht, NULL, start, end);
+}
+
+void
+continuous_agg_invalidate_mat_ht(const Hypertable *raw_ht, const Hypertable *mat_ht, int64 start,
+								 int64 end)
+{
+	ContinuousAggHypertableStatus caggstatus;
+	int32 entry_id;
+	Oid main_table_relid = InvalidOid;
+
+	Assert(raw_ht != NULL);
 
-	if (hypertable_is_distributed(ht))
+	if (mat_ht)
 	{
-		remote_invalidation_log_add_entry(ht, caggstatus, entry_id, start, end);
+		entry_id = mat_ht->fd.id;
+		main_table_relid = mat_ht->main_table_relid;
 	}
 	else
 	{
-		ContinuousAggHypertableStatus caggstatus = ts_continuous_agg_hypertable_status(ht->fd.id);
+		entry_id = raw_ht->fd.id;
+		main_table_relid = raw_ht->main_table_relid;
+	}
 
+	caggstatus = ts_continuous_agg_hypertable_status(entry_id);
+
+	if (hypertable_is_distributed(raw_ht))
+	{
+		remote_invalidation_log_add_entry(raw_ht, caggstatus, entry_id, start, end);
+	}
+	else
+	{
 		switch (caggstatus)
 		{
 			case HypertableIsMaterialization:
-				invalidation_cagg_log_add_entry(ht->fd.id, start, end);
+				invalidation_cagg_log_add_entry(entry_id, start, end);
 				break;
 
 			case HypertableIsRawTable:
-				invalidation_hyper_log_add_entry(ht->fd.id, start, end);
+				invalidation_hyper_log_add_entry(entry_id, start, end);
 				break;
 
 			case HypertableIsMaterializationAndRaw:
```

Comment:
this can not happen since it would hit one of the 2 cases above it

Reply:
We build using the `-Werror=switch` compiler option so we need to test all enumeration values, no?